src/api2/managers/signers-manager.js

```diff
@@ -0,0 +1,134 @@
+import { TransactionBuilder } from '../../base/transaction_builder'
+import { ManageSignerBuilder } from '../../base/operations/manage_signer_builder'
+
+import { NotFoundError } from '../../errors'
+
+const RECOVERY_SIGNER_IDENTITY = 1
+const DEFAULT_SIGNER_IDENTITY = 0
+const DEFAULT_SIGNER_WEIGHT = 1000
+const DEFAULT_SIGNER_ROLE_KEY = 'signer_role:default'
+
+/**
+ * Signers manager.
+ */
+
+export class SignersManager {
+  /**
+   * SignersManager constructor.
+   *
+   * @param {ApiCaller} apiCaller ApiCaller instance to process the requests.
+   */
+  constructor (apiCaller) {
+    this._apiCaller = apiCaller
+  }
+
+  /**
+   * Creates change signers transaction envelope
+   *
+   * @param {object} opts
+   * @param {string} opts.newPublicKey New master signer ID of account
+   * @param {string} opts.sourceAccount Transaction source account ID.
+   * @param {string} [opts.signerToReplace] Specified signer ID to change.
+   * @param {Keypair} opts.signingKeypair Keypair for signing transaction.
+   *
+   * @return {Promise.<string>} Base64-encoded transaction envelope
+   */
+  async createChangeSignerTransaction ({
+    newPublicKey,
+    sourceAccount,
+    signerToReplace,
+    signingKeypair
+  }) {
+    const tx = new TransactionBuilder(sourceAccount)
+    tx.operations = await this._getChangeSignerOperations({
+      newPublicKey, sourceAccount, signerToReplace
+    })
+
+    const txEnv = tx.build()
+    txEnv.sign(signingKeypair)
+
+    return txEnv.toEnvelope().toXDR().toString('base64')
+  }
+
+  async _getChangeSignerOperations ({
+    newPublicKey,
+    sourceAccount,
+    signerToReplace
+  }) {
+    const nonRecoverySigners = await this._getNonRecoverySigners(sourceAccount)
+
+    let operations = []
+    const createSignerOp = await this._getCreateSignerOp(newPublicKey)
+    operations.push(createSignerOp)
+
+    if (nonRecoverySigners.length) {
+      const removeSignerOps = signerToReplace
+        ? this._getRemoveMasterAndCurrentSignerOps(
+          nonRecoverySigners,
+          sourceAccount,
+          signerToReplace
+        )
+        : this._getRemoveAllSignersOps(nonRecoverySigners)
+
+      operations.push(...removeSignerOps)
+    }
+
+    return operations
+  }
+
+  async _getNonRecoverySigners (accountId) {
+    const signers = await this._getSigners(accountId)
+    return signers
+      .filter(signer => signer.identity !== RECOVERY_SIGNER_IDENTITY)
+  }
+
+  async _getSigners (accountId) {
+    try {
+      const endpoint = `/v3/accounts/${accountId}/signers`
+      const { data: signers } = await this._apiCaller.get(endpoint)
+
+      return signers
+    } catch (err) {
+      if (err instanceof NotFoundError) {
+        return []
+      }
+    }
+  }
+
+  async _getCreateSignerOp (newAccountId) {
```

Comment:
a bit confusing that above you use `get` for the method that performs http request, here you use it for method it doesn't. use smth like `make` for methods that will return you operations

Reply:
Renamed